pkg/crypto/key_manager.go

```diff
@@ -0,0 +1,109 @@
+// It leverages private helper functions defined in keys.go for performing these operations,
+// thereby abstracting the underlying cryptographic mechanisms from the rest of the application.
+//
+// Features and Responsibilities:
+// - Manages Libp2p and ECDSA cryptographic keys, including private and public keys.
+// - Supports loading private keys from environment variables or files, with a fallback
+//   to generating new keys if necessary.
+// - Provides functionality to marshal and unmarshal private keys, as well as to convert
+//   keys between different formats (e.g., from Libp2p to ECDSA format, and vice versa).
+// - Offers methods to get hex-encoded representations of keys and to convert public keys
+//   to Ethereum address format.
+// - Ensures thread-safe initialization and access to the cryptographic keys through the
+//   use of the sync.Once mechanism.
+//
+// Usage:
+// To access the KeyManager and its functionalities, use the KeyManagerInstance() function
+// which returns the singleton instance of KeyManager. This instance can then be used to
+// perform various key management tasks, such as retrieving the application's cryptographic
+// keys, converting key formats, and more.
+// Example:
+//     keyManager := crypto.KeyManagerInstance()
+
+var (
+	keyManagerInstance *KeyManager
+	once               sync.Once
+)
+
+// KeyManager holds all the cryptographic entities used in the application.
+type KeyManager struct {
+	Libp2pPrivKey crypto.PrivKey    // Libp2p private key
+	Libp2pPubKey  crypto.PubKey     // Libp2p public key
+	EcdsaPrivKey  *ecdsa.PrivateKey // ECDSA private key
+	EcdsaPubKey   *ecdsa.PublicKey  // ECDSA public key
+	HexPrivKey    string            // Hex-encoded private key
+	HexPubKey     string            // Hex-encoded public key
+	EthAddress    string            // Ethereum format address
+}
+
+// KeyManagerInstance returns the singleton instance of KeyManager, initializing it if necessary.
+func KeyManagerInstance() *KeyManager {
+	once.Do(func() {
+		keyManagerInstance = &KeyManager{}
+		if err := keyManagerInstance.loadPrivateKey(); err != nil {
+			logrus.Fatal("Failed to initialize keys:", err)
+		}
+	})
+	return keyManagerInstance
+}
+
+func (km *KeyManager) loadPrivateKey() (err error) {
+	var keyFile string
+	cfg := config.GetInstance()
+	if cfg.PrivateKey != "" {
+		km.Libp2pPrivKey, err = getPrivateKeyFromEnv(cfg.PrivateKey)
+	} else {
+		keyFile = config.GetInstance().PrivateKeyFile
+		// Check if the private key file exists
+		km.Libp2pPrivKey, err = getPrivateKeyFromFile(keyFile)
+		if err != nil {
+			km.Libp2pPrivKey, err = generateNewPrivateKey(keyFile)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	km.Libp2pPubKey = km.Libp2pPrivKey.GetPublic()
+	// After obtaining the libp2p privKey, convert it to an ECDSA private key
+	km.EcdsaPrivKey, err = libp2pPrivateKeyToEcdsa(km.Libp2pPrivKey)
+	if err != nil {
+		return err
+	}
+	// TODO: do we still need the ecdsa version saved to a file?
```

Comment:
@restevens402 we do so the key can be imported to Metamask if needed - lets do this for now but we may kill this in the future